website/server/models/user/methods.js

```diff
@@ -35,13 +35,39 @@ schema.methods.sendMessage = async function sendMessage (userToReceiveMessage, m
   await Bluebird.all(promises);
 };
 
+/**
+ * Creates a notification based on the input parameters and adds it to the local user notifications array.
+ * This does not save the notification to the database or interact with the database in any way.
+ *
+ * @param  type  The type of notification to add to the user. Possible values are defined in the UserNotificaiton Schema
+ * @param  data  The data to add to the notification
+ */
 schema.methods.addNotification = function addUserNotification (type, data = {}) {
   this.notifications.push({
     type,
     data,
   });
 };
 
+/**
+ * Creates a notification based on the type and data input parameters and saves that new notification
+ * to the database directly using an update statement. The local copy of these users are not updated by
+ * this operation. Use this function when you want to add a notification to a user(s), but do not have
+ * the user document(s) opened.
+ *
+ * @param  userIds An array containing the ids of the users to add this notification to
+ * @param  type  The type of notification to add to the user. Possible values are defined in the UserNotificaiton Schema
+ * @param  data  The data to add to the notification
+ */
+schema.statics.pushNotification = async function pushNotification (userIds, type, data = {}) {
+  let newNotification = new UserNotification({type, data});
```

Comment:
let's validate this to be sure all values are valid, `newNotification.validateSync()`, it returns `undefined` if the doc is valid or the error if it's not, http://mongoosejs.com/docs/api.html#document_Document-validateSync